Add storage-branches feature support for branched storage API (#477)
keboola/mcp-server: integtests/README.md

project at a time, so concurrent CI jobs or overlapping local runs do not corrupt each
other's data.

> > **Important**: The integration tests are **not** tied to specific projects. Any clean
> > Keboola project with a workspace can be used. For local development you only need **two**

Reviewer: ```suggestion
> **Important**: The integration tests are **not** tied to specific Keboola projects but to any  
> Keboola project that has proper features enrolled.
```
AFAIK the workspace is needed only for some test and in general the test suite is generating workspaces on it's behalf

Author: workspaces ARE required for pool projects (FQN/query tests fail without them). Updated the intro to also mention both dedicated branch test projects. Fixed in a9d4097c

Reviewer: then CI/CD must fail as well or?

> projects in the pool (one Snowflake, one BigQuery) plus one old-branches project. The CI

> > pool uses four projects for concurrency, but this is not a requirement for local runs.

Reviewer: I would not mention the CI, it's obvious, CI is living it's own live 

```suggestion
> projects in the pool (one Snowflake, one BigQuery) plus one old-branches project.
```

Author: Fixed in e72d503e

Removed the CI mention.

---

## 1. Setting Up Your Own Test Environment

### Do NOT reuse the CI projects

The CI pool projects are shared across all CI runners. If you point your local `.env` at
them, you will interfere with CI runs: the pool lock mechanism will block or — worse — your
local run may acquire a CI project, crash mid-way, and leave it in a dirty state that

> causes random CI failures.

Reviewer: This is not possible to enforce by user, how you know that you are running on CI project except calling `token/verify` before starting testing on tokens that you have put to pool? I think it's either not needed and expected that CI projects are not reused.

Author: @Matovidlo I don't understand this comment. This is guidance for developers/claude that you should not use existing CI projects (the ones that are listed in the readme) for local testing. To avoid problems that I caused during my testing

Reviewer: I think not using CI project is obvious, we always need to have separate projects and have some instructions how to set them up. It feels like we need a skill for setting up the projects as new developer and how to setup CI projects. Otherwise this is non relevant if there is skill what we should do in order to run integtest and how they behave.

Author: It's empty projects, setting up empty project is obvious. It wasn't obvious to me that I should use my own... It's just a documentation Martine -  the more the better. I don't want to remove this line, it can be helpful (would be for me)

  connection details
- **From SQL**: Open the SQL editor in the workspace and run `SHOW SCHEMAS;` — look for the
  schema starting with `WORKSPACE_`
- **From the API**:
  ```bash
  curl -s -H "X-StorageApi-Token: YOUR_TOKEN" \
    "https://connection.YOUR-STACK.keboola.com/v2/storage/branch/default/workspaces" \
    | python3 -c "import json,sys; [print(f'{w[\"id\"]}: {w[\"connection\"][\"schema\"]}') for w in json.load(sys.stdin)]"
  ```

### `.env` file

Create a `.env` file in the project root:

```dotenv
# Required — pool projects
INTEGTEST_POOL_STORAGE_API_URL=https://connection.europe-west3.gcp.keboola.com
INTEGTEST_STORAGE_TOKENS=<token-project-A> <token-project-B>
INTEGTEST_WORKSPACE_SCHEMAS=<WORKSPACE_for_A> <WORKSPACE_for_B>

# Required — branch storage tests (dedicated projects, not in pool)
INTEGTEST_STORAGE_TOKEN_STORAGE_BRANCHES=<token-for-project-WITH-storage-branches>
INTEGTEST_STORAGE_TOKEN_OLD_BRANCHES=<token-for-project-WITHOUT-storage-branches>

# Optional — second project for multi-client tests
INTEGTEST_STORAGE_TOKEN_PRJ2=<token>
INTEGTEST_WORKSPACE_SCHEMA_PRJ2=<WORKSPACE_XXX>
```

> **Critical**: The order of `INTEGTEST_STORAGE_TOKENS` and `INTEGTEST_WORKSPACE_SCHEMAS`

Reviewer: AFAIK, to run integtests you do not need workspace schemas, either out CI/CD is broken or it's not needed. I would rework it in way that each test creates and deleted the workspace in which the test run should use it (query_data tool integtests)

Author: Yes I agree, but to me this is out of the scope of this PR. Let's handle it when optimising the integtests https://linear.app/keboola/issue/AI-3034/review-integration-tests-ensure-backend-specific-tests-run-on-both I don't want to bloat this PR
